palette/support/src/dispatch.rs

```diff
@@ -1009,9 +1042,16 @@ macro_rules! decl_module {
 		$vis fn $name(
 			$origin: $origin_ty $(, $param: $param_ty )*
 		) -> $crate::dispatch::DispatchResult<$error_type> {
-			{ $( $impl )* }
-			// May be unreachable.
-			Ok(())
+			use $crate::rstd::if_std;
+			if_std! {
+				use $crate::tracing;
+				let span = tracing::span!(tracing::Level::INFO, stringify!($name));
+				let _enter = span.enter();
```

Comment:
hmm I think I read somewhere that these stop timing when they go out of scope? I assume the scoping here is okay and they end when the function impl goes out of scope? I am worried that the `_enter` is created an immediately dropped since it is in a separate scope

Reply:
yeah, it does stay in scope - see here: https://gist.github.com/mattrutherford/8b0395a1e061c4b5f0fdbd39aea70382

give it a try if in doubt: 
```./target/release/substrate --dev --tracing-targets=pallet=info``` 

Values seem to be in expected range.